lib/pbench/server/api/resources/upload_api.py

```diff
@@ -128,35 +159,51 @@ def put(self, controller):
             )
             dataset.add()
         except DatasetDuplicate:
-            self.logger.info("Dataset already exists {}", filename)
+            self.logger.info(
+                "Dataset already exists, user = {}, ctrl = {}, file = {}",
+                username,
+                controller,
+                filename,
+            )
             response = jsonify(dict(message="Dataset already exists"))
             response.status_code = 200
             return response
         except Exception:
-            self.logger.exception("unable to create dataset for {}", filename)
+            self.logger.exception(
+                "unable to create dataset for user = {}, ctrl = {}, file = {}",
+                username,
+                controller,
+                filename,
+            )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
         if tar_full_path.is_file() or md5_full_path.is_file():
             self.logger.error(
-                "Dataset or corresponding md5 file already present on the disc, {}",
-                filename,
+                "Dataset or corresponding md5 file already present on the disc, {}, {}",
+                tar_full_path,
+                md5_full_path,
             )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
-        self.logger.info("Uploading file {} to {}", filename, dataset)
+        self.logger.info(
+            "Uploading file {} (user = {}, ctrl = {}) to {}",
+            username,
+            controller,
+            filename,
+            dataset,
+        )
 
         with tempfile.NamedTemporaryFile(mode="wb", dir=path) as ofp:
-            chunk_size = 4096
             self.logger.debug("Writing chunks")
             hash_md5 = hashlib.md5()
 
             try:
                 while True:
-                    chunk = request.stream.read(chunk_size)
+                    chunk = request.stream.read(self.CHUNK_SIZE)
```

Comment:
Instead of `self.CHUNK_SIZE`, should this be `Upload.CHUNK_SIZE`?  (A la the reference to `ALLOWED_EXTENSION` in `allowed_file()`....)  Ditto line 69.

Reply:
Class variables can be accessed using `self`.

Reply:
Yes...they _can_...but that doesn't mean that they _should_....

For ones which are functionally constants, class attributes are better accessed via the class name -- this makes it clear what their scope and usage is.  (If the values are actually instance attributes which happen to be initialized at the class level, then obviously its appropriate (and presumably necessary) to access them via `self`, but that's not the use case here.)

Beyond that, it would be good to be consistent with the reference in `allowed_file()` (although, I realize that there is no choice there, since that is a static method) -- otherwise, you make the reader work extra hard to establish that the two references reach the same value.